zeroize/src/lib.rs

```diff
@@ -263,6 +263,30 @@ pub trait Zeroize {
 #[allow(drop_bounds)]
 pub trait ZeroizeOnDrop: Drop {}
 
+/// Auto-deref workaround for deriving `ZeroizeOnDrop`.
+#[doc(hidden)]
+pub trait AssertZeroizeOnDrop {
+    fn zeroize_or_on_drop(&mut self);
+}
+
+#[doc(hidden)]
```

Comment:
I’d suggest making these proper sealed traits by placing them in a private module, rather than merely hiding them from the docs. Otherwise they’re still a part of the public API.

Reply:
They need to be part of the public API, otherwise the derive proc-macro can't access them. Some alternatives:
- Rename them further to `__AssertZeroizeOnDrop`.
- Remove them from `zeroize` and let the proc-macro insert them. This has the disadvantage of requiring every `impl` block to add them, which might increase compile time.
- Mark traits as `unsafe`.

Or a combination of these.

Reply:
> otherwise the derive proc-macro can't access them

Aah!

> Rename them further to __AssertZeroizeOnDrop.

Another option would be to put them in a module like `__internal`, which would also let you do `#[doc(hidden)]` in a single place.

> Mark traits as unsafe.

Since the trait doesn't deal strictly with memory safety I'd prefer not to do this.

Reply:
Done!